### PR TITLE
switch to Clap 4.1

### PR DESCRIPTION
**Pull request by stlankes**
The old clap version uses unmaintained crates. This PR should close issue #73, #100, and #110.

### CODE REVIEW

File: src/logging.rs [hunk 1]
```diff
@@ -7,15 +8,52 @@ use std::fs::OpenOptions;
...
+
+#[derive(Copy, Clone, Debug, PartialEq, Eq, PartialOrd, Ord, ValueEnum)]
+pub enum LogLevel {
+	Info,
+	Warn,
+	Debug,
+	Trace,
+	Error,
+	Off,
+}
```
Comment (mkroening) [inline src/logging.rs:36]: Could we use [`log::LevelFilter`](https://docs.rs/log/latest/log/enum.LevelFilter.html) directly?

File: src/delete.rs [hunk 2]
```diff
@@ -59,9 +59,6 @@ pub fn delete_container(project_dir: PathBuf, id: Option<&str>, force: bool) {
 	//Additionally to deleting all the files, we should also delete all remaining processes spawned by the container init process.
 	//However, without cgroup support there is currently no real way to do this as we do not know when (and how) the init process will be killed
 	} else {
-		warn!(
-			"Container `{}` doesn't exists! Skipping deletion...",
-			id.unwrap()
-		);
+		warn!("Container `{}` doesn't exists! Skipping deletion...", id);
```
Comment (mkroening) [inline src/delete.rs:62]:
```suggestion
		warn!("Container `{id}` doesn't exists! Skipping deletion...");
```

**Review by mkroening — APPROVED** [review]
Looks good to me 👍 

I left two suggestions.
Also, I opened https://github.com/hermitcore/runh/pull/167 to update the lockfile separately before updating clap. Ideally, these commits would also be squashed. :)

File: src/logging.rs (hunk 1 above)
Comment (stlankes) [reply src/logging.rs:36]: I think that this isn't possible. Clap's parser doesn't support it and I need to derive with the entry `ValueEnum`.

```
#[derive(Copy, Clone, Debug, PartialEq, Eq, PartialOrd, Ord, ValueEnum)]
#pub enum LogLevel
```

**Pull request merged**